src/executor/nested_loop_join.rs

```diff
@@ -22,40 +22,42 @@ pub struct NestedLoopJoinExecutor {
 impl NestedLoopJoinExecutor {
     #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
     pub async fn execute(self) {
-        // collect all chunks from children
-        let (left_chunks, right_chunks) = async {
-            tokio::try_join!(
-                self.left_child.try_collect::<Vec<DataChunk>>(),
-                self.right_child.try_collect::<Vec<DataChunk>>(),
-            )
-        }
-        .await?;
+        // only support inner and left outer join
```

Comment:
We should also panic on `BoundJoinOperator::RightOuter | BoundJoinOperator::FullOuter` to avoid producing wrong result.